workflows/differentialabundance.nf

```diff
@@ -60,34 +63,54 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
 
 // Info required for completion email and summary
 def multiqc_report = []
-
+//TODO: make lowercase? https://nf-co.re/docs/contributing/guidelines/requirements/workflow_name
 workflow DIFFERENTIALABUNDANCE {
 
     ch_versions = Channel.empty()
 
     //
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
     //
-    INPUT_CHECK (
-        ch_input
-    )
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    //TODO
+    //INPUT_CHECK (
+    //    ch_input
+    //)
+    //ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
 
     //
-    // MODULE: Run FastQC
+    // MODULE: Run ImportMergedCounts
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
+    IMPORTMERGEDCOUNTS (
+        ch_samplesheet,
+        ch_merged_gene_counts
     )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
-
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
+    ch_modified_samplesheet = IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet //file(IMPORTMERGEDCOUNTS.out.ch_modified_samplesheet.first())
+    ch_modified_gene_counts = IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts //file(IMPORTMERGEDCOUNTS.out.ch_modified_gene_counts.first())
+    ch_versions = ch_versions.mix(IMPORTMERGEDCOUNTS.out.versions)
+
+    ch_input = Channel.fromPath(params.contrast_meta)
```

Comment:
If it helps, the way I imagined implementing this was that we'd have the input checker output the things that should 'travel together' as a channel (like in https://github.com/nf-core/modules/pull/2392/files, though appreciate that may not be the one we use).

Then we'd use a cross operation to make a channel with the matrix components plus the split-out contrasts. https://www.nextflow.io/docs/latest/operator.html#cross

Reply:
There might be another operation or two involved to the above to get the right channel format.

Note that the current state of the DESeq2 module is to take a single channel: https://github.com/nf-core/modules/blob/master/modules/nf-core/deseq2/differential/main.nf

Reply:
See also [my tweak to the DESeq2 module](https://github.com/pinin4fjords/nf-core-modules/blob/698d4a446dcd7f7302fa2d48994f6271a201f48b/tests/modules/nf-core/deseq2/differential/main.nf#L59) where I use `combine()` as @ggabernet was suggesting.